### ISSUE TITLE
RCTBridge required dispatch_sync to load RCTAccessibilityManager. This may lead to deadlocks

### ISSUE THREAD

**Issue by correafederico25** [facebook/react-native#42728]
### Description

When building a new project with **React Native 0.73.3** I am getting the warning **RCTBridge required dispatch_sync to load RCTAccessibilityManager. This may lead to deadlocks** as soon as my app starts.

I tried creating a new app with version **0.73.2** and this warning disappears

This my package JSON file:

```
{
  "name": "AwesomeProject3",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start",
    "test": "jest"
  },
  "dependencies": {
    "react": "18.2.0",
    "react-native": "0.73.3"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native/babel-preset": "0.73.20",
    "@react-native/eslint-config": "0.73.2",
    "@react-native/metro-config": "0.73.4",
    "@react-native/typescript-config": "0.73.1",
    "@types/react": "^18.2.6",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.6.3",
    "eslint": "^8.19.0",
    "jest": "^29.6.3",
    "prettier": "2.8.8",
    "react-test-renderer": "18.2.0",
    "typescript": "5.0.4"
  },
  "engines": {
    "node": ">=18"
  }
}
```

### Steps to reproduce

npx react-native@latest init AwesomeProject

### React Native Version

0.73.3

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
No
```

### Stacktrace or Logs

```text
WARN RCTBridge required dispatch_sync to load RCTAccessibilityManager. This may lead to deadlocks
```

### Reproducer

No

### Screenshots and Videos

![Captura de pantalla 2024-01-30 a la(s) 05 10 08](https://github.com/facebook/react-native/assets/53522327/77d4bbf0-b447-4531-ba0f-9069e80910ef)

**Comment by efstathiosntonas** [comment]
Getting the same after upgrading to 0.73.3, @fabOnReact care to explain why you linked the comments above?

**Comment by Jonnboy91** [comment]
Also getting the same after upgrading to 0.73.3 and I don't have the issue with 0.73.2 either.

**Comment by cortinico** [comment]
@cipolleschi can you look into this (potentially related to f39f34ed82997d0595522a285c3cb8693594e718)

**Comment by joeljerushan** [comment]
`"react-native": "0.73.3"` still same

**Comment by kulakowka** [comment]
I have the same issue. 

```js
"react-native": "^0.73.2",
```

**Comment by jomavazquez** [comment]
Me too:

Creating a project now using:
npx react-native init MyProject --template react-native-template-typescript

this downloads 0.73.3 RN version, and then without any modification, I get the error in IOS.

Waiting this new PR approved in 0.73.4 or whatrver. Thank you

**Comment by Titozzz** [comment]
Version 0.73.4 with the fix with be released Monday!
In the meantime you should be safe ignoring that warning.

Sorry for the inconvenience

**Comment by andreyradkevich** [comment]
@Titozzz you can't ignore it , for example by app just doesn't work because of this issue .  Is it planned to release it today ?

**Comment by olivierlesnicki** [comment]
+1

**Comment by ChromeQ** [comment]
@olivierlesnicki This issue is fixed in version 0.73.4 - this issue should be closed.
